sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -80,14 +97,14 @@ static az_result root(az_event_policy* me, az_event event)
       break;
 
     default:
-      // TODO_L Event filtering. We should not ignore events.
+      // TODO_L Event filtering. We should not ignore events. Pipeline should not be sending them
+      // down.
       break;
   }
 
   return ret;
 }
 
-/* TODO_L: Unused, will implement later.
 static az_result faulted(az_event_policy* me, az_event event)
 {
   az_result ret = AZ_ERROR_HFSM_INVALID_STATE;
```

Comment:
Have you tested what happens on the Entry event? I vaguely remember having to return AZ_OK on entry to the client faulted state otherwise it acts as if the transition fails, which bubbles up kinda weird errors.

Reply:
Good point, changed to handle the entry event (not handling exit since it is unrecoverable).